### PR TITLE
[vcpkg] copy tools pdb if they exist.

### LINKED ISSUE #14359: [vcpkg_copy_tools] needs to also copy & clean pdb files
if executables are built with pdb files `vcpkg_copy_tools` needs to also copy the pdb files and clean them.

### PR DESCRIPTION
**Pull request by Neumann-A**
just as the title say. Also remove them if they exists and the tool gets removed from bin

closes #14359

### CODE REVIEW

**Comment by BillyONeal** [comment]
```
[135/151] cmd.exe /C "cd . && D:\downloads\tools\cmake-3.18.4-windows\cmake-3.18.4-win32-x86\bin\cmake.exe -E vs_link_dll --intdir=corelib\src\CMakeFiles\rtabmap_core.dir --rc="C:\PROGRA~2\Windows Kits\10\bin\10.0.19041.0\x64\rc.exe" --mt="C:\PROGRA~2\Windows Kits\10\bin\10.0.19041.0\x64\mt.exe" --manifests  -- "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x86\link.exe"  @CMakeFiles\rtabmap_core.rsp  /out:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\bin\rtabmap_cored.dll /implib:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\lib\rtabmap_cored.lib /pdb:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\bin\rtabmap_cored.pdb /dll /version:0.20 /machine:X86 /nologo    /debug /INCREMENTAL  && cd ."
FAILED: D:/buildtrees/rtabmap/src/1a6c3ea0ca-12abde5708.clean/bin/rtabmap_cored.dll D:/buildtrees/rtabmap/src/1a6c3ea0ca-12abde5708.clean/lib/rtabmap_cored.lib 
cmd.exe /C "cd . && D:\downloads\tools\cmake-3.18.4-windows\cmake-3.18.4-win32-x86\bin\cmake.exe -E vs_link_dll --intdir=corelib\src\CMakeFiles\rtabmap_core.dir --rc="C:\PROGRA~2\Windows Kits\10\bin\10.0.19041.0\x64\rc.exe" --mt="C:\PROGRA~2\Windows Kits\10\bin\10.0.19041.0\x64\mt.exe" --manifests  -- "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x86\link.exe"  @CMakeFiles\rtabmap_core.rsp  /out:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\bin\rtabmap_cored.dll /implib:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\lib\rtabmap_cored.lib /pdb:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\bin\rtabmap_cored.pdb /dll /version:0.20 /machine:X86 /nologo    /debug /INCREMENTAL  && cd ."
LINK Pass 1: command "C:\PROGRA~2\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.27.29110\bin\Hostx64\x86\link.exe @CMakeFiles\rtabmap_core.rsp /out:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\bin\rtabmap_cored.dll /implib:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\lib\rtabmap_cored.lib /pdb:D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\bin\rtabmap_cored.pdb /dll /version:0.20 /machine:X86 /nologo /debug /INCREMENTAL /MANIFEST /MANIFESTFILE:corelib\src\CMakeFiles\rtabmap_core.dir/intermediate.manifest corelib\src\CMakeFiles\rtabmap_core.dir/manifest.res" failed (exit code 1120) with the following output:
octomap.lib(OcTreeNode.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomap.lib(OcTreeNode.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
octomap.lib(Pointcloud.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomap.lib(Pointcloud.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
octomap.lib(AbstractOcTree.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomap.lib(AbstractOcTree.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
octomap.lib(AbstractOccupancyOcTree.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomap.lib(AbstractOccupancyOcTree.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
octomap.lib(ColorOcTree.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomap.lib(ColorOcTree.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
octomath.lib(Vector3.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomath.lib(Vector3.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
octomath.lib(Quaternion.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomath.lib(Quaternion.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
octomath.lib(Pose6D.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Rtabmap.cpp.obj
octomath.lib(Pose6D.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Rtabmap.cpp.obj
   Creating library D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\lib\rtabmap_cored.lib and object D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\lib\rtabmap_cored.exp
LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library
OctoMap.cpp.obj : error LNK2001: unresolved external symbol "protected: static class rtabmap::RtabmapColorOcTree::StaticMemberInitializer rtabmap::RtabmapColorOcTree::RtabmapColorOcTreeMemberInit" (?RtabmapColorOcTreeMemberInit@RtabmapColorOcTree@rtabmap@@1VStaticMemberInitializer@12@A)
D:\buildtrees\rtabmap\src\1a6c3ea0ca-12abde5708.clean\bin\rtabmap_cored.dll : fatal error LNK1120: 1 unresolved externals
```

Doesn't seem related to this PR...

File: scripts/cmake/vcpkg_copy_tools.cmake [hunk 1]
```diff
@@ -34,24 +34,27 @@ function(vcpkg_copy_tools)
...
         message(FATAL_ERROR "SEARCH_DIR ${_vct_SEARCH_DIR} is supposed to be a directory.")
     endif()
 
-    foreach(tool_name ${_vct_TOOL_NAMES})
+    foreach(tool_name IN LISTS _vct_TOOL_NAMES)
         set(tool_path "${_vct_SEARCH_DIR}/${tool_name}${VCPKG_TARGET_EXECUTABLE_SUFFIX}")
-
-        if(EXISTS ${tool_path})
-            file(COPY ${tool_path} DESTINATION ${CURRENT_PACKAGES_DIR}/tools/${PORT})
+        set(tool_pdb "${_vct_SEARCH_DIR}/${tool_name}.pdb")
```
Comment (JackBoosY) [inline scripts/cmake/vcpkg_copy_tools.cmake:44]: We also need to automatically clean them up when using `AUTO_CLEAN`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by PhoebeHui** [comment]
The PR https://github.com/microsoft/vcpkg/pull/14400 would adress the baseline failure.

File: scripts/cmake/vcpkg_copy_tools.cmake (hunk 1 above)
Comment (Neumann-A) [reply scripts/cmake/vcpkg_copy_tools.cmake:44]: Did you miss the change in the other file/function?

Comment (JackBoosY) [reply scripts/cmake/vcpkg_copy_tools.cmake:44]: Oh sorry.

**Review by PhoebeHui — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**